flow/layers/layer.h

```diff
@@ -80,12 +89,54 @@ struct PrerollContext {
   // than to remember the value so that it can choose the right strategy
   // for its |Paint| method.
   bool subtree_can_inherit_opacity = false;
+
+  std::vector<RasterCacheEntry> raster_cached_entries;
 };
 
-class PictureLayer;
-class DisplayListLayer;
-class PerformanceOverlayLayer;
-class TextureLayer;
+struct RasterCacheEntry {
+  RasterCacheEntry(CacheableLayer* layer,
+                   PrerollContext& context,
+                   const SkMatrix& matrix,
+                   unsigned num_child,
+                   bool need_caching = false)
+      : layer(layer),
+        matrix(matrix),
+        raster_cache(context.raster_cache),
+        mutators_stack(context.mutators_stack),
+        cull_rect(context.cull_rect),
+        color_space(context.dst_color_space),
+        num_child_entries(num_child),
+        need_caching(need_caching) {}
+
+  explicit RasterCacheEntry(CacheableLayer* layer,
+                            PrerollContext& context,
+                            const SkMatrix& matrix,
+                            bool need_caching = false)
+      : RasterCacheEntry(layer, context, matrix, 0, need_caching) {}
+
+  void UpdateRasterCacheEntry(CacheableLayer* layer,
```

Comment:
This is only used to update the layer to point to the cacheable child. There is no reason to set any other values here than to update the layer. Since the layer is exposed, the caller can just modify the field without needing this method.